### PR TITLE
[tests-only][full-ci]add test coverage for `/app/open` endpoint

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
Add acceptance test for `/app/open` end point. Covered cases for:
- `/app/open?file_id={file_id}&app_name={app_name}`
- `/app/open?file_id={file_id}`

## Related Issue

- https://github.com/owncloud/ocis/issues/9682

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Review by SagarGi — COMMENTED** [review]
LGTM :+1: if CI is green

File: .drone.star [hunk 1]
```diff
@@ -158,7 +158,20 @@ config = {
             "suites": [
                 "cliCommands",
             ],
+        },
+        "apiAppProvider": {
```
Comment (saw-jan) [inline .drone.star:162]:
```suggestion
        "apiWopi": {
```

File: tests/acceptance/features/apiAppProvider/appProvider.feature
Comment (saw-jan) [inline tests/acceptance/features/apiAppProvider/appProvider.feature:1]: `apiCollaboration/wopi.feature`

File: .drone.star [hunk 2]
```diff
@@ -880,9 +895,10 @@ def localApiTestPipeline(ctx):
                                      (waitForClamavService() if params["antivirusNeeded"] else []) +
                                      (waitForEmailService() if params["emailNeeded"] else []) +
                                      (ocisServer(storage, params["accounts_hash_difficulty"], deploy_type = "federation", extra_server_environment = params["extraServerEnvironment"]) if params["federationServer"] else []) +
+                                     (collaborationService(extra_environment = params["extraCollaborationEnvironment"]) if params["collaborationServiceNeeded"] else []) +
```
Comment (saw-jan) [inline .drone.star:898]:
```suggestion
                                     (collaborationService(params["extraCollaborationEnvironment"]) if params["collaborationServiceNeeded"] else []) +
```

File: tests/config/drone/app-registry.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+app_registry:
+  mimetypes:
+    - mime_type: application/vnd.oasis.opendocument.text
+      extension: odt
+      name: FakeOffice
+      description: FakeOffice text document
```
Comment (saw-jan) [inline tests/config/drone/app-registry.yaml:6]:
```suggestion
      name: OpenDocument
      description: OpenDocument text document
```

File: tests/acceptance/config/behat.yml [hunk 4]
```diff
@@ -414,6 +414,13 @@ default:
         - TagContext:
         - TrashbinContext:
 
+    apiAppProvider:
+      paths:
```
Comment (saw-jan) [inline tests/acceptance/config/behat.yml:418]: Please, move it above cli suites

File: .drone.star [hunk 5]
```diff
@@ -2894,6 +2911,63 @@ def waitForClamavService():
         ],
     }]
 
+def fakeOffice():
```
Comment (saw-jan) [inline .drone.star:2914]: reuse it in other existing pipelines as well

File: .drone.star [hunk 6]
```diff
@@ -2894,6 +2911,63 @@ def waitForClamavService():
...
+        {
+            "name": "wait-for-fakeoffice",
+            "image": OC_CI_WAIT_FOR,
+            "commands": [
+                "wait-for -it fakeoffice:8080 -t 300",
+            ],
+        },
+    ]
+
+def collaborationService(extra_environment = {}):
```
Comment (saw-jan) [inline .drone.star:2934]: reuse it in other existing pipelines as well.

File: tests/acceptance/features/apiAppProvider/appProvider.feature [hunk 7]
```diff
@@ -0,0 +1,51 @@
+Feature: App Provider
```
Comment (saw-jan) [inline tests/acceptance/features/apiAppProvider/appProvider.feature:1]:
```suggestion
Feature: collaboration (wopi)
```

File: tests/acceptance/features/apiAppProvider/appProvider.feature [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  As a user
+  I want to access files with collaboration service apps
+  So that I can collaborate with other users
```
Comment (saw-jan) [inline tests/acceptance/features/apiAppProvider/appProvider.feature:4]: I find it hard to properly write user story for API feature

File: tests/acceptance/features/apiAppProvider/appProvider.feature [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        "type": "object",
+        "required": [
+          "app_url",
+          "method",
+          "form_parameters"
+        ],
+        "properties": {
+          "app_url": {
+            "type": "string",
+            "pattern": "^https:\\/\\/(.*?)\\?WOPISrc=http%3A%2F%2F(.*?)%3A9300%2Fwopi%2Ffiles%2F[a-fA-F0-9]{64}$"
```
Comment (saw-jan) [inline tests/acceptance/features/apiAppProvider/appProvider.feature:25]:
```suggestion
            "pattern": "^.*\\?WOPISrc=.*wopi%2Ffiles%2F[a-fA-F0-9]{64}$"
```

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: .drone.star [hunk 10]
```diff
@@ -158,7 +158,20 @@ config = {
...
+            ],
             "skip": False,
+            "collaborationServiceNeeded": True,
+            "extraCollaborationEnvironment": {
+                "COLLABORATION_APP_NAME": "FakeOffice",
+                "COLLABORATION_APP_ADDR": "http://fakeoffice:8080",
+            },
+            "extraServerEnvironment": {
+                "GATEWAY_GRPC_ADDR": "0.0.0.0:9142",
+            },
```
Comment (saw-jan) [inline .drone.star:174]: Is it possible to list the required services and use it in the pipeline services
```suggestion
            "extraServerEnvironment": {
                "GATEWAY_GRPC_ADDR": "0.0.0.0:9142",
            },
            "extraServices": [
                fakeOffice,
            ],
```

Comment (saw-jan) [reply .drone.star:169]: will more changes in this PR. can be done in coming PRs

File: .drone.star [hunk 11]
```diff
@@ -154,11 +154,25 @@ config = {
...
+        },
+        "extraServices": [],
```
Comment (saw-jan) [inline .drone.star:171]:
```suggestion
        },
```

File: .drone.star [hunk 12]
```diff
@@ -154,11 +154,25 @@ config = {
...
         "cli": {
             "suites": [
                 "cliCommands",
             ],
-            "skip": False,
```
Comment (saw-jan) [inline .drone.star:161]: :arrows_counterclockwise:

File: tests/acceptance/features/apiCollaboration/wopi.feature [hunk 13]
```diff
@@ -0,0 +1,51 @@
+Feature: collaboration (wopi)
+  As a user
+  I want to access files with collaboration service apps
+  So that I can collaborate with other users
+
+
+  Scenario Outline: open file with .odt extension with collaboration app
```
Comment (saw-jan) [inline tests/acceptance/features/apiCollaboration/wopi.feature:7]:
```suggestion
  Scenario Outline: open file with .odt extension
```

**Review by saw-jan — APPROVED** [review]
LGTM

**Pull request merged**